File: src/codes/hardcoded-variability/php/ksrc.php
<?php

class ReportGenerator {
  public function generatePdfReport(string $data): void {
    echo "=== レポート開始 ===\n";
    echo "[PDF形式] {$data}\n";
    echo "=== レポート終了 ===\n";
  }

  public function generateCsvReport(string $data): void {
    echo "=== レポート開始 ===\n";
    echo "[CSV形式] {$data}\n";
    echo "=== レポート終了 ===\n";
  }

  public function generateHtmlReport(string $data): void {
    echo "=== レポート開始 ===\n";
    echo "<html><body>{$data}</body></html>\n";
    echo "=== レポート終了 ===\n";
  }
}

File: src/codes/hardcoded-variability/py/ksrc.py
class ReportGenerator:
    def generate_pdf_report(self, data: str):
        print("=== レポート開始 ===")
        print(f"[PDF形式] {data}")
        print("=== レポート終了 ===")

    def generate_csv_report(self, data: str):
        print("=== レポート開始 ===")
        print(f"[CSV形式] {data}")
        print("=== レポート終了 ===")

    def generate_html_report(self, data: str):
        print("=== レポート開始 ===")
        print(f"<html><body>{data}</body></html>")
        print("=== レポート終了 ===")
